sdk/Managed/src/Microsoft.WindowsAzure.MobileServices/Table/Sync/MobileServiceSyncContext.cs

```diff
@@ -370,6 +372,17 @@ private Task ExecuteOperationAsync(MobileServiceTableOperation operation, JObjec
 
                 try
                 {
+                    if (MobileServiceTableOperationKind.Insert != operation.Kind)
+                    {
+                        // Refresh the item's version, in case it changed while we were awaiting the item lock
+                        var localStoreItem = await this.Store.LookupAsync(operation.TableName, operation.ItemId);
```

Comment:
Yes, I completely agree and we need a more flushed out concept of client conflicts for these scenarios. 

I think the delegate needs some new methods on it, like a updateReceivedForItem:, and let the client code respond & update their local copy as needed.  For operations that were blocked, they would also need to trigger something similar once they get the lock again and know how to resolve the "local conflict".  

I'll need to review my earlier notes & discuss internally on what our plans are on this area again.  Please continue to let us know the pain points you are seeing so we can prioritize correctly.